src/server/pps/server/pipeline_controller.go

```diff
@@ -635,50 +578,20 @@ func (op *pipelineOp) scaleDownPipeline() (retErr error) {
 // if errorState {
 //   return op.restartPipeline("entered error state")
 // }
-//
-// Like other functions in this file, restartPipeline takes responsibility for
-// retrying and eventually failing op's pipeline if restartPipeline can't
-// restart it.
 func (op *pipelineOp) restartPipeline(reason string) error {
-	var errCount int
-	if err := backoff.RetryNotify(func() error {
-		if op.rc != nil && !op.rcIsFresh() {
-			// delete old RC, monitorPipeline goro, and worker service
-			if err := op.deletePipelineResources(); err != nil {
-				return err
-			}
-		}
-		// create up-to-date RC
-		if err := op.createPipelineResources(); err != nil {
-			return err
+	if op.rc != nil && !op.rcIsFresh() {
+		// delete old RC, monitorPipeline goro, and worker service
+		if err := op.deletePipelineResources(); err != nil {
+			return newRetriableError(err, "error deleting resources for restart")
 		}
-		return op.setPipelineState(pps.PipelineState_PIPELINE_RESTARTING, "")
-	}, backoff.NewInfiniteBackOff(), func(err error, d time.Duration) error {
-		if errCount++; errCount >= maxErrCount {
-			return err
-		}
-		log.Errorf("PPS master: error restarting pipeline %q: %v; retrying in %v", op.name, err, d)
-		return nil
-	}); err != nil {
-		return op.failPipeline(fmt.Sprintf("could not restart after %d attempts: %v", errCount, err))
 	}
-	return errors.Errorf("restarting pipeline %q: %v", op.name, reason)
-}
-
-// failPipeline fails op's pipeline. failPipeline is an error-handling codepath,
-// so it's guaranteed to return an error (typically wrapping 'reason', though if
-// the restart process fails that error will take precendence) so that callers
-// can use it like so:
-//
-// if errorState {
-//   return op.failPipeline("entered error state")
-// }
-//
-// Like other functions in this file, failPipeline takes responsibility for
-// retrying.
-func (op *pipelineOp) failPipeline(reason string) error {
-	if err := op.m.a.setPipelineFailure(op.opClient.Ctx(), op.name, reason); err != nil {
-		return errors.Wrapf(err, "error failing pipeline %q", op.name)
+	// create up-to-date RC
+	if err := op.createPipelineResources(); err != nil {
+		return errors.Wrap(err, "error creating resources for restart")
+	}
+	if err := op.setPipelineState(pps.PipelineState_PIPELINE_RESTARTING, ""); err != nil {
+		return errors.Wrap(err, "error restarting pipeline")
 	}
-	return errors.Errorf("failing pipeline %q: %v", op.name, reason)
+
+	return errors.Errorf("restarting pipeline %q: %v", op.name, reason)
```

Comment:
Nit: format string (no `%v`)

Reply:
I guess this was there before my change, but switched to %s. Sorry about the others, I could have sworn I did a pass to make sure none were left.